### PR TITLE
NEW Add gauge release and dispatching gha-auto-tag.

### PR DESCRIPTION
**Pull request by GuySartorelli**
This allows for a dispatchable patch tagging workflow, which means we can remove the need for `contents:write` in the main ci.yml.

## Issue
- https://github.com/silverstripe/gha-ci/issues/137

### CODE REVIEW

**Comment by ChloeSartorelli** [comment]
Checked this works as expected in a workflow identical to [the one being added](https://github.com/silverstripe/module-standardiser/pull/66) in the following scenarios:

- [latest_local_sha isn't latest commit on that branch](https://github.com/GuySartorelli/issue-template-playground/actions/runs/10154131074/job/28078738976)
- [sha is latest commit but not on minor branch](https://github.com/GuySartorelli/issue-template-playground/actions/runs/10154140289/job/28078765254)
- [on minor branch but no patch tags](https://github.com/GuySartorelli/issue-template-playground/actions/runs/10154159466/job/28078816864)
- [has tag on that branch but nothing to release](https://github.com/GuySartorelli/issue-template-playground/actions/runs/10154173109/job/28078853460)
- [has stuff to release](https://github.com/GuySartorelli/issue-template-playground/actions/runs/10154199584/job/28078918143) which made https://github.com/GuySartorelli/issue-template-playground/releases/tag/1.0.1 (note I wasn't using PRs which is why there's no specific changes noted in the description)

File: README.md [hunk 1]
```diff
@@ -17,17 +17,41 @@ jobs:
...
-The tag to create e.g. 1.2.3
+### Latest local sha
+Required if not skipping gauge release.
+
+The result of `$(git rev-parse HEAD)`.
+
+`latest_local_sha: f22dbc6ec6118096c8ccccee1ca0074bfb2f2291`
+
+### Skip gauge release
+Whether to skup gauging the release. Gauging the release will only allow tagging if all of the following are true:
```
Comment (emteknetnz) [inline README.md]:
```suggestion
Whether to skip gauging the release. Gauging the release will only allow tagging if all of the following are true:
```

File: README.md [hunk 2]
```diff
@@ -17,17 +17,41 @@ jobs:
...
+
+- The branch this action is run against is a patch branch (e.g. `1.2`)
+- The `latest_local_sha` input matches the `github.sha` GitHub actions variable
+- The `latest_local_sha` input matches the current latest commit sha on this branch
+- There's an existing stable semver patch tag for this branch already (i.e. for branch `1.2` there must be a `1.2.<digit>` tag)
+- There are commits on the branch that warrant a new patch release and weren't included in the latest patch release for this branch
+
+Gauging the release also identifies the correct next patch tag, and uses that to tag the release.
```
Comment (emteknetnz) [inline README.md]: Mention what the default value for skip_gauge_release is

File: action.yml [hunk 3]
```diff
@@ -1,49 +1,282 @@
...
+  latest_local_sha:
+    description: The latest local sha. Used to gauge the release
+    required: false
+    type: string
+  skip_gauge_release:
+    description: If true, skip straight to tagging the release
+    required: false
+    default: false
+    type: boolean
+  # Note: the following inputs should usually be left as default if using guage release
```
Comment (emteknetnz) [inline action.yml]:
```suggestion
  # Note: the following inputs should usually be left as default if using gauge release
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: action.yml (hunk 3 above)
Comment (ChloeSartorelli) [reply action.yml]: Done

File: README.md (hunk 2 above)
Comment (ChloeSartorelli) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (ChloeSartorelli) [reply README.md]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**